jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -164,13 +164,17 @@ private static Decoder createDecoder(ObjectMapper objectMapper, ObjectMapper cbo
     }
 
     private static ImmutableList<ConnectionSpec> createConnectionSpecs(boolean enableGcmCipherSuites) {
-        return ImmutableList.of(
-            new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
-                    .tlsVersions(TlsVersion.TLS_1_2)
-                    .cipherSuites(enableGcmCipherSuites
-                            ? CipherSuites.allCipherSuites()
-                            : CipherSuites.fastCipherSuites())
-                    .build(),
-            ConnectionSpec.CLEARTEXT);
+        ConnectionSpec.Builder encryptedSpecs = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                .tlsVersions(TlsVersion.TLS_1_2)
+                .cipherSuites(enableGcmCipherSuites
+                        ? CipherSuites.allCipherSuites()
+                        : CipherSuites.fastCipherSuites());
+
+        // Works around https://github.com/square/okhttp/issues/3173 until OkHttp 3.7 upgrade
```

Comment:
I did try bumping to 3.8 (and 3.7), however then I ran into classpath issues (was getting NoSuchMethodException).